test/config-next/orphan-finder.json

```diff
@@ -6,9 +6,9 @@
   },
 
   "tls": {
-    "caCertFile": "test/grpc-creds/minica.pem",
-    "certFile": "test/grpc-creds/orphan-finder.boulder/cert.pem",
-    "keyFile": "test/grpc-creds/orphan-finder.boulder/key.pem"
+    "caCertFile": "../../test/grpc-creds/minica.pem",
```

Comment:
Why do these paths need to be prefixed with `../../` in this config file but not the others? e.g.:

https://github.com/letsencrypt/boulder/blob/83aafd18842e093483d6701b92419ca8f7f1855b/test/config-next/sa.json#L8-L12

https://github.com/letsencrypt/boulder/blob/83aafd18842e093483d6701b92419ca8f7f1855b/test/config-next/ocsp-updater.json#L16-L20

https://github.com/letsencrypt/boulder/blob/83aafd18842e093483d6701b92419ca8f7f1855b/test/config-next/admin-revoker.json#L5-L9

Reply:
This is because the Go integration tests are run using `go test`. In `go test`, tests are run with a current directory equal to their build path. Since the orphan-finder config is only used in the Go integration tests, using the `../../` makes things work, and doesn't break other contexts.

The other configs are used when starting up boulder components via `test/startservers.py`. When that runs, the current directory is always the root of the Boulder repo.

An alternative would be to have the Go test `Chdir` to the Boulder repo root before running any commands. If we go this route the main thing I'd have to figure out is whether that could affect any other test cases running in parallel.

Reply:
Thanks for explaining, that makes sense :+1: 

> The other configs are used when starting up boulder components via test/startservers.py. When that runs, the current directory is always the root of the Boulder repo.

I can imagine a future where we'll want to start other components from the Go integration tests. I think it makes sense to try and land a generic way to start a program from the root of the Boulder repo in the common Go integration test code so that we can write the configurations the same way.

I'd be happy to +1 this with the relative paths in `orphan-finder.json` if you wanted to file an issue for a follow-up.

Reply:
https://github.com/letsencrypt/boulder/issues/4508

Reply:
Thanks!